v2/git-initializer/main.go

```diff
@@ -102,10 +102,62 @@ func gitCheckout() error {
 		auth = publicKeys
 	}
 
+	var worktree *git.Worktree
+	if event.Worker.Git.Commit == "" && event.Worker.Git.Ref == "" {
+		worktree, err = simpleCheckout(gitConfig.CloneURL, auth)
+	} else {
+		worktree, err = complexCheckout(
+			gitConfig.CloneURL,
+			event.Worker.Git.Commit,
+			event.Worker.Git.Ref,
+			auth,
+		)
+	}
+	if err != nil {
+		return err
+	}
+
+	// Initialize submodules if configured to do so
+	if event.Worker.Git.InitSubmodules {
+		err = initSubmodules(worktree)
+	}
```

Comment:
It doesn't matter whether the simple or complex scenario played out up to this point. Again, some logic that used to be here, just moved.